app/models/user_password.rb

```diff
@@ -1,14 +1,67 @@
 # frozen_string_literal: true
 
 class UserPassword < ActiveRecord::Base
-  validates :user_id, presence: true
+  MAX_PASSWORD_LENGTH = 200
+  TARGET_PASSWORD_ALGORITHM =
+    "$pbkdf2-#{Rails.configuration.pbkdf2_algorithm}$i=#{Rails.configuration.pbkdf2_iterations},l=32$"
+  PASSWORD_SALT_LENGTH = 16
+
+  attr_reader :password # required to build an instance of this model with password attribute without backing column
```

Comment:
does anything actually read this? If not, I'd be tempted to remove it, and also remove the `@password=...` line. 

The less we keep raw passwords around, the better!

(if it's actually needed then no worries, ignore this comment)

Reply:
I see, yeah. I think the strategy of `User#password` is probably the way to go. Having a 'public' method on the class which could potentially return the plaintext password feels kinda risky. Ofc, not an issue by itself, but it increases the surface area for developers to make mistakes.

I hear you on `User#password` implementation being a bit 'weird' though. Maybe it should return `nil` rather than an empty string?